drivers/gles3/storage/config.cpp

```diff
@@ -169,8 +169,26 @@ Config::Config() {
 
 	//Adreno 3xx Compatibility
 	const String rendering_device_name = String::utf8((const char *)glGetString(GL_RENDERER));
-	//TODO: Check the number between 300 and 399(?)
-	adreno_3xx_compatibility = (rendering_device_name.left(13) == "Adreno (TM) 3");
+	if (rendering_device_name.left(13) == "Adreno (TM) 3") {
+		adreno_3xx_compatibility = true;
+
+		// ignore driver version 331+
+		const String gl_version = String::utf8((const char *)glGetString(GL_VERSION));
+		// Adreno 3xx examples (https://opengles.gpuinfo.org/listreports.php):
+		// ===========================================================================
+		// OpenGL ES 3.0 V@84.0 AU@ (CL@)
+		// OpenGL ES 3.0 V@127.0 AU@ (GIT@I96aee987eb)
+		// OpenGL ES 3.0 V@140.0 AU@ (GIT@Ifd751822f5)
+		// OpenGL ES 3.0 V@251.0 AU@08.00.00.312.030 (GIT@Ie4790512f3)
+		// OpenGL ES 3.0 V@269.0 AU@ (GIT@I109c45a694)
+		// OpenGL ES 3.0 V@331.0 (GIT@35e467f, Ice9844a736) (Date:04/15/19)
+		// OpenGL ES 3.0 V@415.0 (GIT@d39f783, I79de86aa2c, 1591296226) (Date:06/04/20)
+		// OpenGL ES 3.0 V@0502.0 (GIT@09fef447e8, I1fe547a144, 1661493934) (Date:08/25/22)
+		String driver_version = gl_version.get_slice("V@", 1).get_slice(" ", 0);
+		if (driver_version.is_valid_float() && driver_version.to_float() >= 331.0) {
+			adreno_3xx_compatibility = false;
```

Comment:
We check `adreno_3xx_compatibility` to disable GPUParticles https://github.com/godotengine/godot/blob/a7aa8356b9288fb6de1cef2980465703c63102ea/drivers/gles3/storage/particles_storage.cpp#L125

This PR will re-enable GPUParticles on certain driver versions. Can you check that GPUParticles don't cause a crash on devices with the newer driver version?

Reply:
Unfortunately I can't test it:
https://github.com/godotengine/godot/issues/92662#issuecomment-2161199477

Reply:
Actually, looking at https://github.com/godotengine/godot/pull/88816 which introduces the check, it seems that the same GPU was tested as the one reported in https://github.com/godotengine/godot/issues/92662.

That means we need to split the `adreno_3xx_compatibility` into 2 parts. I would rename it into two bools:
1. `disable_particles_workaround`: This should use the old logic and should be `true` for all 3XX devices. 
2. `flip_xy_workaround`: This should use the new logic from this PR

Reply:
I would also like to wait for the answer, 2 to 3 days. Whether `Adreno 3xx` and `V@415.0` driver also has problems with GPUParticles.


---------------------------------------

**confirmed:**
```
		// OpenGL ES 3.0 V@127.0 AU@ (GIT@I96aee987eb)
		// OpenGL ES 3.0 V@140.0 AU@ (GIT@Ifd751822f5)
		// OpenGL ES 3.0 V@269.0 AU@ (GIT@I109c45a694)
```

???	->	`// OpenGL ES 3.0 V@331.0 (GIT@35e467f, Ice9844a736) (Date:04/15/19)`

**no crash and no flip_xy bug:** https://github.com/godotengine/godot/issues/92662#issuecomment-2144031734
`// OpenGL ES 3.0 V@415.0 (GIT@d39f783, I79de86aa2c, 1591296226) (Date:06/04/20)`

Reply:
Sounds good. I have reached out to a contact at Qualcomm to see if they know exactly what driver version this was fixed in